feat: native ARM builds + updated workflow architecture

sudo-kraken opened the pull request:
- Replace QEMU emulation with native ARM runners for performance reasons,
- Standardise all workflows with cache-warming, deterministic timestamps,
- Added more container metadata while maintaining clean registries

gauthier-th commented:
@M0NsTeRRR review welcome 😄

M0NsTeRRR commented on .github/workflows/ci.yml:
```diff
@@ -44,111 +52,109 @@ jobs:
+
+      - name: Deterministic timestamp
+        id: ts
+        run: echo "now=$(date -u +%Y-%m-%dT%H:%M:%SZ)" >> "$GITHUB_OUTPUT"
```
Use `echo "TIMESTAMP=$(git log -1 --pretty=%ct)" >> "$GITHUB_OUTPUT"` to get docker reproductible build https://docs.docker.com/build/ci/github-actions/reproducible-builds/#git-commit-timestamps

And we can add `ARG SOURCE_DATE_EPOCH` in the `Dockerfile`

sudo-kraken replied:
Ah, I went with `date -u` just to have a clean UTC timestamp, but tying it to git log makes more sense to tie the build date to the commit.

M0NsTeRRR commented on .github/workflows/ci.yml:
```diff
@@ -44,111 +52,109 @@ jobs:
-          outputs: |
-            type=image,push-by-digest=true,name=fallenbagel/jellyseerr,push=true
-            type=image,push-by-digest=true,name=ghcr.io/${{ env.OWNER_LC }}/jellyseerr,push=true
+            BUILD_DATE=${{ steps.ts.outputs.now }}
```
Use `SOURCE_DATE_EPOCH=${{ steps.ts.outputs.TIMESTAMP }}`
and configure output with `rewrite-timestamp=true` [rewrite-timestamp doc](https://github.com/moby/buildkit/blob/master/docs/build-repro.md#source_date_epoch)

M0NsTeRRR commented on .github/workflows/release.yml:
```diff
@@ -1,120 +1,182 @@
+
       - name: Release
+        id: release
+        uses: cycjimmy/semantic-release-action@v4
```
We should remove the dependency inside the `package.json` and use [extra_plugins](https://github.com/cycjimmy/semantic-release-action?tab=readme-ov-file#extra_plugins) ?

gauthier-th replied:
No no we don't use it locally

M0NsTeRRR commented on .github/workflows/ci.yml:
```diff
@@ -44,111 +52,109 @@ jobs:
+            COMMIT_TAG=${{ github.sha }}
+            BUILD_VERSION=develop
+            BUILD_DATE=${{ steps.ts.outputs.now }}
+          labels: |
+            org.opencontainers.image.source=${{ github.server_url }}/${{ github.repository }}
+            org.opencontainers.image.revision=${{ github.sha }}
+            org.opencontainers.image.version=develop
+          tags: |
+            fallenbagel/jellyseerr:develop
+            fallenbagel/jellyseerr:${{ github.sha }}
+            ghcr.io/${{ env.OWNER_LC }}/jellyseerr:develop
+            ghcr.io/${{ env.OWNER_LC }}/jellyseerr:${{ github.sha }}
```
We can use [docker/metadata-action](https://github.com/docker/metadata-action)

M0NsTeRRR commented:
I didn’t re-add some comments I previously made in #1905 that apply here.

M0NsTeRRR commented:
Also you can looks into [zizmor tool](https://github.com/zizmorcore/zizmor), I haven't run it yet but it's a security static analysis tool to find vulnetability in workflows

sudo-kraken commented:
> Yup looks good, you forgot to push ?

I have pushed the others on PR-1905, still working on the release based ci flows.

sudo-kraken commented:
@M0NsTeRRR I have pushed the updates to clean up package.json, removed the local @semantic-release/git and @semantic-release/changelog since we already inject them via workflow extra_plugins. 

Adjusted Dockerfile to streamline build, removed the hard-coded OCI labels from the Dockerfile and shifted them into the metadata-action step.

Updated metadata steps to add standard OCI labels and ensure consistency across release/preview/CI builds.

I've been doing this between jobs today so please can you re-review and ensure I didnt miss anything obvious.

M0NsTeRRR commented on .github/workflows/release.yml:
```diff
@@ -1,120 +1,204 @@
+            ghcr.io/${{ env.OWNER_LC }}/jellyseerr
+          tags: |
+            type=raw,value=${{ needs.semantic-release.outputs.new_release_version }}
+          labels: |
```
Looks like we only need to overwrite `created` ?

https://github.com/docker/metadata-action/blob/master/src/meta.ts#L536

M0NsTeRRR commented on .github/workflows/release.yml:
```diff
@@ -1,120 +1,204 @@
+        uses: cycjimmy/semantic-release-action@v4
+        with:
+          extra_plugins: |
+            @semantic-release/git
```
does it make sense to pin the major at least ?

https://github.com/cycjimmy/semantic-release-action?tab=readme-ov-file#extra_plugins

M0NsTeRRR commented on package.json:
```diff
@@ -208,28 +206,12 @@
     "plugins": [
       "@semantic-release/commit-analyzer",
       "@semantic-release/release-notes-generator",
-      [
```
From what I read here https://github.com/cycjimmy/semantic-release-action?tab=readme-ov-file#extra_plugins

We have to keep this configuration.

M0NsTeRRR reviewed (commented):
I’ve added some comments. After this review, could you test the release process in your fork? We want to avoid breaking anything with all the recent changes.

As for migrating from merge JSON to cache, I’m neutral—I’ve never thought about using a cache for this, but I don’t see any problems with it either, except maybe with cache poisonning.
